### PR TITLE
CAMEL-10679

### PR DESCRIPTION
**Pull request by marcusmesserkewill**
Producer does not populate attachments from response.

I would think of refactoring the whole part of the response handling.
In my opinion it would be best to handle this the same way as the
BasicMessageFilter does this for the request message.
This way anyone could implement custom behaviour if neccessary.
I would have done it, but i didnt want to break anybodys code, so i just
implemented it the same way the headers are populated.
Feel free to contact me about the change.

### CODE REVIEW

File: components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceConfiguration.java [hunk 1]
```diff
@@ -401,5 +403,22 @@ public boolean isAllowResponseHeaderOverride() {
...
+    public boolean isAllowResponseAttachmentOverride() {
+        return allowResponseAttachmentOverride;
+    }
 
+    /**
+     * Option to override soap response attachments in in/out exchange with attachments from the actual service layer.
+     * If the invoked service appends or rewrites the soap attachments this option when set to true, allows the modified
+     * soap attachments to be overwritten in in/out message attachments
+     * 
+     * @param allowResponseAttachmentOverride
```
Comment (davsclaus) [inline components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceConfiguration.java]: Can you put the description on the same line as `@param`

File: components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceConfiguration.java [hunk 2]
```diff
@@ -401,5 +403,22 @@ public boolean isAllowResponseHeaderOverride() {
     public void setAllowResponseHeaderOverride(boolean allowResponseHeaderOverride) {
         this.allowResponseHeaderOverride = allowResponseHeaderOverride;
     }
+    /**
```
Comment (davsclaus) [inline components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceConfiguration.java]: imho better to just have javadoc on the setter

File: components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceProducer.java [hunk 3]
```diff
@@ -139,7 +149,18 @@ private void populateMessageHeaderFromResponse(Message message, SoapHeader soapH
 
         }
     }
-
+    /**
+     * Populates message attachments from soap response attachments 
+     * @param inOrOut {@link Message}
+     * @param soapMessage {@link SoapMessage}
+     */
+    private void populateMessageAttachmentsFromResponse(Message inOrOut, Iterator<Attachment> attachments) {
```
Comment (davsclaus) [inline components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceProducer.java]: Looks like you may have checkstyle issues. Can you run 

   mvn compile -Psourcecheck

**Comment by oscerd** [comment]
I'll wait for the fixes and I'll test and merge. Thanks @marcusmesserkewill and @davsclaus

File: components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceProducer.java (hunk 3 above)
Comment (marcusmesserkewill) [inline components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceProducer.java]: I made those fixes you requested, even though i did not implemented the soap header part, i just added the attachment population. ;)

But anyway, let me know if you have any other change requests.

Regards Marcus

**Comment by marcusmesserkewill** [comment]
I just saw, the adoc is auto generated, so this probaply doesnt need to be committed i guess.
Everything fine now? Or shall i remove the commit of the .adoc?
Please note i renamed the unit test method in my last commit.

**Comment by davsclaus** [comment]
yeah the table in the adoc files are auto generated

**Comment by davsclaus** [comment]
@oscerd will get this merged

**Comment by oscerd** [comment]
yeah, I'll take care of this later today :-)

**Comment by oscerd** [comment]
Thanks. The PR has been merged. Can you please close this one? @davsclaus do we need to backport to 2.18.x too?

**Comment by marcusmesserkewill** [comment]
From my point of view, it would be really nice, if you backport it to 2.18.x.

I have a project where i need this behaviour, right now i am using a modified version of 2.18.2-SNAPSHOT.
Where i initially made this pull request for.

**Comment by marcusmesserkewill** [comment]
Closed as requested.
Thanks a lot. :)

**Comment by oscerd** [comment]
The main problem in backporting is that we have a new option on the endpoint and usually we avoid this situation in minor release. By the way lets wait for @davsclaus

**Comment by davsclaus** [comment]
@oscerd i think in this case its safer as its a seldom used component and its a change that dont change any existing behavior.

**Comment by oscerd** [comment]
Great, I'll backport then :-)

**Comment by marcusmesserkewill** [comment]
Very nice. Thank you.
For when is the 2.18.2 scheduled?
I somewhere saw its scheduled for beginning of 2017.

**Comment by oscerd** [comment]
Probably later this month.

**Comment by oscerd** [comment]
@marcusmesserkewill do you think you can still re-open the other PR for 2.18.x? I see a lot of conflicts.

**Comment by marcusmesserkewill** [comment]
i can do a new PR for 2.18.x if you want.
(Because for the master i changed the method names, to be more precise to its matter.

**Comment by oscerd** [comment]
yes, @marcusmesserkewill if you can

**Pull request closed without merging**